gridcomps/History/Sampler/MAPL_TrajectoryMod_smod.F90

```diff
@@ -721,7 +763,15 @@
             sec = hms_2_s(this%Epoch)
             j1 = j0 + int(sec, kind=ESMF_KIND_I8)
             jx0 = real ( j0, kind=ESMF_KIND_R8)
-            jx1 = real ( j1, kind=ESMF_KIND_R8)
+            if (this%use_NWP_1_file == 1) then            
+               ! IODA case:  Epoch + 1 second : for ioda location index recover to match observation files
+               ! this works because all IODA files has same end <= Epoch
+               ! and we use 1 file only
+               jx1 = real ( j1 + 1, kind=ESMF_KIND_R8)
+            else
+               ! normal case: strict design with cutoff time at Epoch
+               jx1 = real ( j1, kind=ESMF_KIND_R8)
+            end if
```

Comment:
As above - invert the logic so that an else clause is not needed here.

I'm also starting to wonder if "1" is not magic but is actually meant as a counter of something.  If that is the case maybe the problem is in the name `use_NWP_1_file`?   Maybe that needs a "num_" prefix or something like that?

Reply:
Actually,  use_NPW_1_file is true (1) or false (0). 
j1 is 6 hr, and j1+1 is 6 hr 1 sec for bisect purpose.   I think the IODA input was generated by R2D2 python code, which did not follow a clean [ is, ie ) python rule to cut the observation time.  So we have to do a special treatment here at the end of 6 hr Epoch.